s2sml/torch_s2s_dataset.py

```diff
@@ -0,0 +1,374 @@
+from datetime import timedelta
+import numpy as np
+import pandas as pd
+import xarray as xr
+import torch
+from torch.utils.data import Dataset
+
+
+class S2SDataset(Dataset):
+    
+    """
+    Class instantiation for file lists of cesm/era5 train, validate, and test data.
+    
+    Args:
+        week (int): lead time week (1, 2, 3, 4, 5, or 6).
+        variable (str): variable (tas2m, prsfc, tas2m_anom, or prsfc_anom).
+        norm (str): normalization. Defaults to zscore. Also use None, minmax, or negone.
+        region (str): region method used. Defaults 'fixed' uses one region. 'random' changes regions.
+        minv (float): minimum value for normalization. Defaults to None.
+        maxv (float): maximum value for normalization. Defaults to None.
+        mnv (float): mean value for normalization. Defaults to None.
+        stdv (float): standard deviation value for normalization. Defaults to None.
+        lon0 (float): bottom left corner of 'fixed' region (0 to 360). Defaults to None.
+        lat0 (float): bottom left corner of 'fixed' region (-90 to 90). Defaults to None.
+        dxdy (float): number of grid cells for 'fixed' or 'random' region. Defaults to 32.
+        feat_topo (boolean): use terrian heights (era5) as feature. Defaults True.
+        feat_lats (boolean): use latitudes as feature. Defaults True.
+        feat_lons (boolean): use longitudes as feature. Defaults True.
+        startdt (str): datetime start. Defaults to 1999-02-01.
+        enddt (str): datetime end. Defaults to 2021-12-31.
+        homedir (str): home directory. Defaults to /glade/scratch/molina/.
+        
+    Returns:
+        list of cesm and era5 files for train/val/test sets.
+    
+    """
+    
+    def __init__(self, week, variable, norm='zscore', region='fixed',
+                 minv=None, maxv=None, mnv=None, stdv=None, lon0=None, lat0=None, dxdy=32,
+                 feat_topo=True, feat_lats=True, feat_lons=True, 
+                 startdt='1999-02-01', enddt='2021-12-31', homedir='/glade/scratch/molina/'):
```

Comment:
@mariajmolina should we make the startdt and enddt based on seasons? 
Ex: `startdt='1999-11-01, enddt='2021-02-28'`

Reply:
These are just default values -- feel free to test using seasons / dates you want to use. I am going to keep the full range because it is just bias correction in this application.